helper/chart_data.js

```diff
@@ -2263,3 +2271,296 @@ const createSolutionsArray = async function (data) {
         console.log(err);
     }
 }
+
+
+//Report for entity level report in assessments
+exports.entityLevelReportData = async function (data) {
+
+    return new Promise(async function (resolve, reject) {
+       
+        //group the data based on completed date   
+        let groupedSubmissionData = await groupArrayByGivenField(data, "completedDate");
+
+        let completedDateKeys = Object.keys(groupedSubmissionData);
+
+        //Sort the date in ascending order
+        let sortedData = completedDateKeys.sort();
+
+        let threshold = config.druid.no_of_assessment_submissions_threshold ? config.druid.no_of_assessment_submissions_threshold : default_no_of_assessment_submissions_threshold;
+
+        if(typeof threshold !== "number"){
+            throw new Error("no_of_assessment_submissions_threshold value should be integer");
+        }
+
+        if (sortedData.length > threshold) {
+            console.log("entered");
+            sortedData = [sortedData[0], sortedData.length - 2, sortedData.length - 1];
```

Comment:
@deepakr123 - Does not work with threshold more than 3

Reply:
resolved